File: src/main/java/com/paragon/api/util/Wrapper.java
package com.paragon.api.util;

import net.minecraft.client.Minecraft;

public interface Wrapper {

    Minecraft mc = Minecraft.getMinecraft();

    default boolean nullCheck() {
        return mc.player == null || mc.world == null;
    }

    default boolean isHovered(float x, float y, float width, float height, int mouseX, int mouseY) {
        return mouseX >= x && mouseX <= x + width && mouseY >= y && mouseY <= y + height;
    }

}

File: src/main/kotlin/com/paragon/util/Wrapper.kt
package com.paragon.util

import net.minecraft.client.Minecraft

interface Wrapper {

    val minecraft: Minecraft
        get() = Minecraft.getMinecraft()

    fun isHovered(x: Float, y: Float, width: Float, height: Float, mouseX: Int, mouseY: Int): Boolean = mouseX >= x && mouseX <= x + width && mouseY >= y && mouseY <= y + height

}